### PR TITLE
Added Popover and TransactionProgress component

### PR DESCRIPTION
**Pull request by deamme**
Commit changes:
- Added IconClose component
- Updated IconAttention to be more flexible
- Updated Countdown component to be more flexible
- Updated SidePanel to use IconClose component
- Added Alert as subcomponent to Info
- Added ProgressBar and TransactionProgress components and documentation
- Added pages for ProgressBar and TransactionProgress components
- Added Popover component and updated Transactionprogress to reflect changes

Closes https://github.com/aragon/aragon-ui/issues/220 and https://github.com/aragon/aragon-ui/issues/199

### CODE REVIEW

File: src/components/TransactionProgress/TransactionProgress.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+          >
+            <CloseButton type="button" onClick={handleClose}>
+              <IconClose />
+            </CloseButton>
+            <Wrapper>
+              <Text size="large" weight="bold">
+                Pending transaction
+              </Text>
+              <ContentWrapper>
+                <Text smallcaps={true} weight="bolder" color="#707070">
```
Comment (sohkai) [inline src/components/TransactionProgress/TransactionProgress.js]: We could use `theme.textSecondary` here (it's the same color).

File: src/components/TransactionProgress/TransactionProgress.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+                  Estimated time:
+                </Text>
+                <Countdown
+                  removeDaysAndHours={true}
```
Comment (sohkai) [inline src/components/TransactionProgress/TransactionProgress.js]: Small nitpick: no need to use the `={true}`

File: src/components/TransactionProgress/TransactionProgress.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+  font-size: 16px;
+  font-weight: 700;
+  color: #21aae7;
+  text-decoration: none;
+`
+
+export default class TransactionProgress extends React.Component {
+  static propTypes = {
+    top: PropTypes.string,
+    left: PropTypes.string,
```
Comment (sohkai) [inline src/components/TransactionProgress/TransactionProgress.js]: We should also add props to control the `endTime`, `progress`, and `transactionHash`.

File: gallery/src/pages/PageTransactionProgress.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+              slow="true"
+              top={top}
+              left={left}
+              handleClose={() => this.setState({ hide: true })}
+            />
+          )}
+          <Container>
+            <Wrapper>
+              <Button
+                onClick={e =>
```
[the remark this answers is not in the document]
Comment (deamme) [reply gallery/src/pages/PageTransactionProgress.js]: Must be something with your version of react-spring or something? It animates to 1 opacity for me.

File: src/components/ProgressBar/ProgressBar.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+const ProgressBar = ({ color, progress }) => (
+  <Spring native from={{ progressValue: 0 }} to={{ progressValue: progress }}>
+    {({ progressValue }) => (
+      <Main>
+        <Base>
+          <Progress
+            color={color}
+            style={{
+              width: progressValue.interpolate(t => `${t * 100}%`),
```
[the remark this answers is not in the document]
Comment (deamme) [reply src/components/ProgressBar/ProgressBar.js]: Haha, my bad

File: src/components/Info/IconInfo.js [hunk 6]
```diff
@@ -55,6 +56,14 @@ IconInfo.propTypes = {
 
 const Action = props => <IconInfo icon={<Attention />} {...props} />
 
+const Alert = props => (
+  <IconInfo
+    background={'rgba(255, 68, 93, 0.0614)'}
```
[the remark this answers is not in the document]
Comment (deamme) [reply src/components/Info/IconInfo.js]: I don't understand this. Should we add a new color ourselves somehow?

File: src/components/TransactionProgress/TransactionProgress.js [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+import { IconClose } from '../../icons'
+import ProgressBar from '../ProgressBar/ProgressBar'
+
+const Card = styled(animated.div)`
+  background: #ffffff;
+  border: 1px solid #e6e6e6;
+  border-radius: 3px;
+  box-shadow: 0 4px 4px 0 rgba(0, 0, 0, 0.06);
+  position: absolute;
+  z-index: 999;
```
[the remark this answers is not in the document]
Comment (deamme) [reply src/components/TransactionProgress/TransactionProgress.js]: I think the component is meant to use a Popover component underneath which should be above everything else as default. Although we should provide with a z-index option.

File: src/components/TransactionProgress/TransactionProgress.js (hunk 3 above)
Comment (deamme) [reply src/components/TransactionProgress/TransactionProgress.js]: My bad

File: src/components/TransactionProgress/TransactionProgress.js [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+import { Text, SafeLink, Countdown, Info, theme } from '@aragon/ui'
+import { Transition, animated } from 'react-spring'
+
+import { springs } from '../../utils/styles'
+
+import { IconClose } from '../../icons'
+import ProgressBar from '../ProgressBar/ProgressBar'
+
+const Card = styled(animated.div)`
+  background: #ffffff;
```
[the remark this answers is not in the document]
Comment (deamme) [reply src/components/TransactionProgress/TransactionProgress.js]: I don't think so. It's white in Figma

**Comment by sohkai** [comment]
Btw, I was also seeing this on `npm run build`, but I'm not really sure what's causing it:

screen shot 2018-10-09 at 4 33 05 pm

File: src/components/TransactionProgress/TransactionProgress.js (hunk 8 above)
[the remark this answers is not in the document]
Comment (sohkai) [reply src/components/TransactionProgress/TransactionProgress.js]: But this background is white too...?

File: gallery/src/pages/PageTransactionProgress.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (sohkai) [reply gallery/src/pages/PageTransactionProgress.js]: Not sure, I purged and reinstalled but still having the opacity problem. `npm ls react-spring` gives me `react-spring@5.9.2`.

File: src/components/Info/IconInfo.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (sohkai) [reply src/components/Info/IconInfo.js]: We can use `color('rgb(255, 68, 93)')` to make a local color variable, and then use [`.alpha()`](https://github.com/aragon/aragon-ui/blob/master/src/components/Radio/RadioListItem.js#L85) as necessary to get different alpha channels so we don't have to copy the rgb string three times.

File: src/components/TransactionProgress/TransactionProgress.js (hunk 8 above)
[the remark this answers is not in the document]
Comment (deamme) [reply src/components/TransactionProgress/TransactionProgress.js]: Sorry, I looked at the Dark UI..

File: src/components/Info/IconInfo.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (deamme) [reply src/components/Info/IconInfo.js]: I see, thank you

**Comment by sohkai** [comment]
Can confirm https://github.com/aragon/aragon-ui/pull/212/commits/cdd6ee9710b6c5e4143ec81a1f66a757102e9f15#diff-f5eebddeafb17884215204484008fd4eL4 fixed the build warnings :).

File: src/components/TransactionProgress/TransactionProgress.js [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+                  Estimated time:
+                </Text>
+                <Countdown removeDaysAndHours end={endTime} />
+              </ContentWrapper>
+              <ProgressBar color={theme.accent} progress={progress} />
+              <FooterWrapper slow={slow}>
+                {slow && (
+                  <Info.Alert>Slow transaction. Retry with more gas</Info.Alert>
+                )}
+                <Link href={`https://etherscan.io/tx/${transactionHash}`}>
```
[the remark this answers is not in the document]
Comment (sohkai) [reply src/components/TransactionProgress/TransactionProgress.js]: That would also work, assuming we don't need to display the transaction hash :).

**Comment by deamme** [comment]
@sohkai I've added a Popover component now which the TransactionProgress component is using in its example. Let me know if this fixed the issue..

**Comment by deamme** [comment]
Closes https://github.com/aragon/aragon-ui/issues/220 and https://github.com/aragon/aragon-ui/issues/199

**Comment by deamme** [comment]
Closing in favor of https://github.com/aragon/aragon-ui/pull/225

**Pull request closed without merging**